types/test.ts

```diff
@@ -20,10 +20,10 @@ function testSelector() {
 
   const foo: string = selector({foo: 'bar'});
 
-  // typings:expect-error
+  // $ExpectError
```

Comment:
Yeah that makes sense. I would prefer to use the `// @ts-expect-error` directive. If we need to support 3.5 for redux-toolkit, we probably need to keep the `// $ExpectError` directive though.

The main reason that I upgraded from `typings-tester` was for compatibility reasons with later versions of TypeScript _and_ for somewhat selfish development reasons. The error messages from `dtslint` were very helpful in making sure I covered all the edge cases correctly in the new typings. Seeing a descriptive error in the output was a lot easier than having to just see "error @ line ###".

I don't mind downgrading now, if you think it's important. But if we do want to support things < 3.9 we'll need some sort of tooling to help with that.

Reply:
One other note, I think the main reason that something like `dtslint` is useful is that it tests for type compatibility across a range of TS versions where the `// @ts-expect-error` only works for the installed version, right?

I think that would be a compelling reason to actually keep something like `dtslint` around, right?

Reply:
Over on redux toolkit, we use a test matrix that just runs `tsc` on the files with each tested TypeScript version.
https://github.com/reduxjs/redux-toolkit/blob/next/.github/workflows/tests.yml

The nice thing about `// @ts-expect-error` is the usage in the editor. If there is an error where one is expected, everything is fine. If the error is absent, you get squiggly lines. So you can just look into the editor to spot where your test breaks, you don't even need to run the tests.

Especially for writing the tests, that's a big thing.

Reply:
Oh, that's a good idea! I think we can do something similar with TravisCI here right?

If I have the go-ahead to do that, I don't think there are any differences between 3.5 and 3.9 (in regards to this library) so we _could_ just say that this only supports 3.5 and above while only testing 3.9 + using the `// @ts-expect-error` comments.

Basically, we can validate that 3.5, 3.6, 3.7, and 3.8 work with these typings (we know they do because of all the tests we've been running in this PR) but only setup automation to check 3.9 and above. I guess we could say this is "verified again 3.5-3.8, and actively tested against 3.9+".

I'm super new to OSS library maintenance, so I'm down for whatever you guys suggest. You're the experts haha.

Reply:
Or I can still do the matrix and use the `// $ExpectError` directive for 3.5+ and `// @ts-expect-error` for the new typings. Either way seems good to me. But the removal of the testing library seems like a more "future-proof" solution.

Reply:
A build matrix would be possible with Travis, but since Travis CI is pretty much kicking out open source projects by restricting build times & becoming more & more unreliable, it would probably be a good idea to switch over to GH actions anyways.

As for testing older TS versions than 3.9... I'd be fine not to, since I don't expect any bigger changes coming up in the near future and the current state has already been tested for a long time. @markerikson what do you think about this?

Reply:
If @markerikson is good with it, I'll do the switch over to GH actions here too. It's the way of the future anyway, right? 😆 

I could also totally break that up into another PR. Just let me know what you would prefer.

Reply:
> As for testing older TS versions than 3.9... I'd be fine not to, since I don't expect any bigger changes coming up in the near future and the current state has already been tested for a long time.

That would also make the testing a lot easier because you wouldn't need to do the swapping of the test comments or maintain separate type files. I was looking at the way you guys did it in the toolkit and that last bit of the job looked a little gnarly with all the `sed` stuff.